### DISCUSSION TITLE
How do I add user details and edit profile options in the drop-down of the logged-in user's menu?

### DISCUSSION THREAD

**Discussion by dushmanta05** [filamentphp/filament#11174] [Help]
### Package

Panel builder

### Package Version

v3.0.0

### How can we help you?

Hello community!

In my dashboard panel, I'm trying to add three things in the drop-down user menu.

- View profile:
When clicked it'll take him to the profile detail page to show the details of the user who is logged in

- Edit Profile:
Edit profile option to change the details of the user.

- Password Reset
To reset the password of logged-in user.

I've added the labels and icons, but I'm just struggling to create the functionality that will enable these features when clicking on them.

Example of how I've enabled the label inside `Filament/AdminPanelProvider.php`

```php
->userMenuItems([
    MenuItem::make()
       ->label('Edit Profile')
       ->url('')
       ->icon('heroicon-o-pencil-square')
])
```

So could anyone please help me with this? I appreciate your effort and thank you for your valuable time.

**Comment by huiyang** [comment]
If you function is resource related, eg: for edit user profile, it should be related to UserResource, then can reuse the resource page:

``` php
->userMenuItems([
      MenuItem::make()
          ->label('Edit Profile')
          ->url(fn (): string => \App\Filament\Resources\UserResource::getUrl('index'))
```

or if it is not resource related, you can create a custom page 

``` php
->userMenuItems([
      MenuItem::make()
          ->label('Edit Profile')
          ->url(fn (): string =>  \App\Filament\Pages\CustomPage::getUrl())
```

for more details:
https://filamentphp.com/docs/3.x/panels/navigation#customizing-the-user-menu
